auth.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from abc import ABCMeta, abstractmethod, abstractproperty
+from functools import wraps
+import hashlib
+import os.path
+import json
+import re
+
+from flask import request, Response
+
+WOTT_CREDENTIALS_PATH = '/opt/wott/credentials'
+WOTT_SCREENLY_CREDENTIAL_NAME = 'screenly_credentials'
+
+
+class Auth(object):
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def authenticate(self):
+        """
+        Let the user authenticate himself.
+        :return: a Response which initiates authentication.
+        """
+        pass
+
+    @abstractproperty
+    def is_authenticated(self):
+        """
+        See if the user is authenticated for the request.
+        :return: bool
+        """
+        pass
+
+    def authenticate_if_needed(self):
+        """
+        If the user performing the request is not authenticated, initiate authentication.
+        :return: a Response which initiates authentication or None if already authenticated.
+        """
+        if not self.is_authenticated:
+            return self.authenticate()
+
+    def update_settings(self, current_password):
+        """
+        Submit updated values from Settings page.
+        :param current_password: the value of "Current Password" field or None if empty.
+        :return:
+        """
+        pass
+
+    @property
+    def template(self):
+        """
+        Get HTML template and its context object to be displayed in Settings page.
+        :return: (template, context)
+        """
+        pass
+
+
+class NoAuth(Auth):
+    display_name = 'Disabled'
+    name = ''
+    config = {}
+
+    def is_authenticated(self):
+        return True
+
+    def authenticate(self):
+        pass
+
+    def check_password(self, password):
+        return True
+
+
+class BasicAuth(Auth):
```

Comment:
This seems to be the same code as in https://github.com/Screenly/screenly-ose/pull/1162 and I guess will be removed after requested by you rebase.